sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/UnionEstimation.scala

```diff
@@ -70,8 +68,30 @@ object UnionEstimation {
       None
     }
 
-    val unionOutput = union.output
+    val newMinMaxStats = computeMinMaxStats(union)
+    val newNullCountStats = computeNullCountStats(union)
+    val newAttrStats = {
+      val updatedNullCountStats = newNullCountStats.keys.map { key =>
```

Comment:
This is my last comment; in this access pattern, It seems we don't need to convert the null stats into `AttributeMap` in `computeNullCountStats`. How about rewriting it so that they return `Seq[Atttribute, ColumnStat]`? Then, we can create a newAttrStats like this;
```
   val newAttrStats = {
      val baseStats = AttributeMap(newMinMaxStats)
      val overwriteStats = newNullCountStats.map { case attrStat @ (attr, stat) =>
        baseStats.get(attr).map { baseStat =>
          attr -> baseStat.copy(nullCount = stat.nullCount)
        }.getOrElse(attrStat)
      }
      AttributeMap(newMinMaxStats ++ overwriteStats)
    }
```
See[ the commit](https://github.com/apache/spark/commit/bdba39fa649c0219c9e4c01bbab06065b03de7c8) for more details.

Reply:
Thanks. I have updated the code based on the commit.